modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/loader/parser/BaseMuleSdkExtensionModelParser.java

```diff
@@ -44,7 +46,8 @@ protected <T> T getParameter(ComponentAst ast, String paramName) {
    * @return an {@link Optional} for the parameter's value
    */
   protected <T> Optional<T> getOptionalParameter(ComponentAst ast, String paramName) {
-    return ofNullable(getParameter(ast, paramName));
+    return ofNullable(ast.getParameter(DEFAULT_GROUP_NAME, paramName)).map(ComponentParameterAst::<T>getValue)
```

Comment:
how does this change the behavior of all the other parsers? IIRC, I originally wrote this method similar to this and then changed it to its current form, don't remember exactly why. Anyway, if this does make sense, then collapse the two map() into one.